Sources/Conduit/Networking/URLSessionClient.swift

```diff
@@ -132,6 +133,11 @@ public struct URLSessionClient: URLSessionClientType {
 
             logger.verbose("Finished processing request through middleware pipeline ✓")
 
+            guard modifiedRequest.url != nil else {
```

Comment:
I'm curious how this is reachable. It seems like the switch-case handling should be exhaustive, either hitting an error or setting the url. I guess the value of the request might not be nil but the url of the request might be? Do you have a sense of where that value is coming from? Do we have control over making the returned request not allow an empty url? That still might not be the final solution though; I'm just trying to understand how the returned request can have a nil url...that seems like more of the root problem to me.

Reply:
In the `refreshBearerTokenWithin` method in `Auth.swift` file we are deliberately making the request URL as `nil`.
`var noOpRequest = URLRequest(url: noOpURL)
  noOpRequest.url = nil`
This is needed for refreshing the token where we just need the token present in the request regardless of the URL is nil or has value.

We discuss over a call about this if needed.

Reply:
@plflanagan I that is happening [here](https://github.com/mindbody/Conduit/blob/main/Sources/Conduit/Auth/Auth.swift#L72)

Reply:
@ruchiupgade-mb if we make this change, will it break the `refreshBearerTokenWithin` ? 
i.e. 
would this success completion block ever get called?
``` 
sessionClient.begin(request: noOpRequest) { data, response, _ in
   if let token: BearerToken = middleware.tokenStorage.tokenFor(client: middleware.clientConfiguration,
                                                                authorization: middleware.authorization),
   token.isValid {
       completion(.value(token))
   }
   else {
       completion(.error(OAuth2Error.clientFailure(data, response)))
   }
}
```

Reply:
Ah I see here ^ the error is not checked `{ data, response, _ in` 

so this block will fire as expected

Reply:
So technically right now, this completion block is not firing at all anywhere....Which may be the root to many other issues!

Reply:
My thought is this started happening with iOS 15 changes to URL Session https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-15-release-notes